src/language/dafnyInstallation.ts

```diff
@@ -142,40 +162,49 @@ async function cloneAllNecessaryDlls(configuredPath: string): Promise<string> {
   const runtimeconfigjson = '.runtimeconfig.json';
   const depsjson = '.deps.json';
   const dls = dlsName + dll;
-  if(configuredPath.endsWith(dls)) {
-    const installationDir = path.dirname(configuredPath);
-    // copy all the files from installationDir to a new folder installationDir/vscode
-    const vscodeDir = path.join(installationDir, 'vscode');
-    // If the folder does not exist, create it
-    if(!await existsAsync(vscodeDir)) {
-      await mkdirAsync(vscodeDir);
+  if(!configuredPath.endsWith(dls)) {
+    return configuredPath;
+  }
+  const installationDir = path.dirname(configuredPath);
+  // copy all the files from installationDir to a new temporary folder
+  const vscodeDir = path.join(os.tmpdir(), await fs.promises.mkdtemp('vscode-dafny-dlls-'));
```

Comment:
Could you just register one callback to delete this directory and everything in it here (using `recursive = true` in the call to `rmDir`), rather than every intermediate directory separately? Then you don't need the complexity of threading that through `ensureDirExists`.